sdk/cognitivelanguage/azure-ai-language-conversations/azure/ai/language/conversations/_serialization.py

```diff
@@ -0,0 +1,2006 @@
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+#
+# --------------------------------------------------------------------------
+
+# pylint: skip-file
+
+from base64 import b64decode, b64encode
+import calendar
+import datetime
+import decimal
+import email
+from enum import Enum
+import json
+import logging
+import re
+import sys
+import codecs
+try:
+    from urllib import quote  # type: ignore
+except ImportError:
+    from urllib.parse import quote  # type: ignore
+import xml.etree.ElementTree as ET
+
+import isodate
+
+from typing import Dict, Any, cast, TYPE_CHECKING
+
+from azure.core.exceptions import DeserializationError, SerializationError, raise_with_traceback
+
+_BOM = codecs.BOM_UTF8.decode(encoding='utf-8')
+
+if TYPE_CHECKING:
+    from typing import Optional, Union, AnyStr, IO, Mapping
+
+class RawDeserializer:
+
+    # Accept "text" because we're open minded people...
+    JSON_REGEXP = re.compile(r'^(application|text)/([a-z+.]+\+)?json$')
+
+    # Name used in context
+    CONTEXT_NAME = "deserialized_data"
+
+    @classmethod
+    def deserialize_from_text(cls, data, content_type=None):
+        # type: (Optional[Union[AnyStr, IO]], Optional[str]) -> Any
+        """Decode data according to content-type.
+
+        Accept a stream of data as well, but will be load at once in memory for now.
+
+        If no content-type, will return the string version (not bytes, not stream)
+
+        :param data: Input, could be bytes or stream (will be decoded with UTF8) or text
+        :type data: str or bytes or IO
+        :param str content_type: The content type.
+        """
+        if hasattr(data, 'read'):
+            # Assume a stream
+            data = cast(IO, data).read()
+
+        if isinstance(data, bytes):
+            data_as_str = data.decode(encoding='utf-8-sig')
+        else:
+            # Explain to mypy the correct type.
+            data_as_str = cast(str, data)
+
+            # Remove Byte Order Mark if present in string
+            data_as_str = data_as_str.lstrip(_BOM)
+
+        if content_type is None:
+            return data
+
+        if cls.JSON_REGEXP.match(content_type):
+            try:
+                return json.loads(data_as_str)
+            except ValueError as err:
+                raise DeserializationError("JSON is invalid: {}".format(err), err)
+        elif "xml" in (content_type or []):
+            try:
+
+                try:
+                    if isinstance(data, unicode):  # type: ignore
+                        # If I'm Python 2.7 and unicode XML will scream if I try a "fromstring" on unicode string
+                        data_as_str = data_as_str.encode(encoding="utf-8")  # type: ignore
+                except NameError:
+                    pass
+
+                return ET.fromstring(data_as_str)
```

Comment:
Bandit in CI is flagging some lines in this file. I talked with Laurent and Izzy and we will ignore these for now and fix in the generator. You can add a comment `# nosec` on each line to ignore.

```suggestion
                return ET.fromstring(data_as_str)  # nosec
```

Reply:
can we modify this file?
and what does this ` #nosec ` means?

Reply:
Yes, you can modify it. Bandit is a tool we run in CI to sniff out security issues. `# no sec` will ignore the flagged issue. This code comes straight from `msrest` which is a library we created to handle serialization/deserialization. I'm letting the generator folks decide how to handle the issue, but they said we're safe to ignore for now.